src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -94,9 +97,19 @@ export const setDataFromUrl: ActionCreator<ThunkAction<
       })
     : null;
 
-  if (entityId && entityId !== activeEntityId) {
-    dispatch(updateEnsObject(entityId));
-    dispatch(fetchSidebarPayload());
+  if (entityId) {
+    if (entityId !== activeEntityId) {
+      dispatch(updateEnsObject(entityId));
+    }
+    dispatch(loadSidebar());
+    const storedState = entityViewerStorageService.getGeneralState();
+    entityViewerStorageService.updateGeneralState(
+      set(
+        'activeGenomeId',
+        genomeIdFromUrl,
+        set(`activeEnsObjectIds.${genomeIdFromUrl}`, entityId, storedState)
+      )
+    );
```

Comment:
Like in the previous PR about sidebar state, why can't we update just the fragment we are interested in? Why do we need to get the whole state here?

Reply:
I am not quite clear about how to update just the fragment. Is it possible to share a snippet?

Reply:
With lodash:

```js
const obj = { foo: { foo: { foo: 'foo' }  } };
merge(obj, {foo: { foo: { bar: 'bar' } }}); // adding a new property

// => obj is now  { foo: { foo: { foo: 'foo', bar: 'bar' }  } }

merge(obj, {foo: { foo: { foo: 'updated_foo' } }}); // updating existing property

// => obj is now  { foo: { foo: { foo: 'updated_foo', bar: 'bar' }  } }
```


Equally:

With stored state defined as `{ activeEnsObjectIds: { human: 'brca2' } }`:

```js
entityViewerStorageService.updateGeneralState({ activeEnsObjectIds: { human: 'fry' } })
```

should update the state to `{ activeEnsObjectIds: { human: 'fry' } }`

Reply:
Thanks! I've updated it. I hope I remember it next time.